backstack/src/commonMain/kotlin/com/slack/circuit/backstack/BackStack.kt

```diff
@@ -93,6 +93,14 @@ public interface BackStack<R : Record> : Iterable<R> {
    */
   public fun restoreState(screen: Screen): Boolean
 
+  /**
+   * Whether the back stack contains the given [record].
+   *
+   * @param includeSaved Whether to also check if the record is contained by any saved back stack
+   *   state. See [saveState].
+   */
+  public fun containsRecord(record: Record, includeSaved: Boolean): Boolean
```

Comment:
Ideally this would be `record: R` but the `R` type is erased for some reason. I'm not entirely sure why that is tbh, as the `push` function works just fine. 🤔

Reply:
Can you expand on what you mean by "erased"? I'm able to run all the `RetainedTest` tests with this patch that uses the generic

```patch
Index: backstack/src/commonMain/kotlin/com/slack/circuit/backstack/SaveableBackStack.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/backstack/src/commonMain/kotlin/com/slack/circuit/backstack/SaveableBackStack.kt b/backstack/src/commonMain/kotlin/com/slack/circuit/backstack/SaveableBackStack.kt
--- a/backstack/src/commonMain/kotlin/com/slack/circuit/backstack/SaveableBackStack.kt	(revision 7c7336e7d153ed6dca7dd22a41ca7975a0a51ae0)
+++ b/backstack/src/commonMain/kotlin/com/slack/circuit/backstack/SaveableBackStack.kt	(date 1710216396972)
@@ -127,7 +127,7 @@
     return false
   }
 
-  override fun containsRecord(record: BackStack.Record, includeSaved: Boolean): Boolean {
+  override fun containsRecord(record: Record, includeSaved: Boolean): Boolean {
     // If it's in the main entry list, return true
     if (record in entryList) return true
 
Index: backstack/src/commonMain/kotlin/com/slack/circuit/backstack/BackStack.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/backstack/src/commonMain/kotlin/com/slack/circuit/backstack/BackStack.kt b/backstack/src/commonMain/kotlin/com/slack/circuit/backstack/BackStack.kt
--- a/backstack/src/commonMain/kotlin/com/slack/circuit/backstack/BackStack.kt	(revision 7c7336e7d153ed6dca7dd22a41ca7975a0a51ae0)
+++ b/backstack/src/commonMain/kotlin/com/slack/circuit/backstack/BackStack.kt	(date 1710216363885)
@@ -99,7 +99,7 @@
    * @param includeSaved Whether to also check if the record is contained by any saved back stack
    *   state. See [saveState].
    */
-  public fun containsRecord(record: Record, includeSaved: Boolean): Boolean
+  public fun containsRecord(record: R, includeSaved: Boolean): Boolean
 
   @Stable
   public interface Record {

```

Reply:
I tried exactly that and Kotlin kept erasing the `R` type to `Nothing` at callsites. I was questioning my past 15 years as a dev tbh 😆 

I'll try on another computer.

Reply:
So the code above works for normal callsites, just not `NavigableCircuitContent` as it erases the `Record` type. Fixed by making `NavigableCircuitContent` generic